### PR TITLE
Add build infrastructure with dummy program bodies

### PR DESCRIPTION
**Pull request by netch80**
Add build infrastructure with dummy program bodies.

### CODE REVIEW

File: src/processor_cxx11/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+set(PROJECT processor_cxx11)
```
Comment (oleksiibidnichenko) [inline src/processor_cxx11/CMakeLists.txt]: this is a redundant variable

File: src/processor_cxx11/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+project(${PROJECT})
+
+set(SOURCES main.cxx)
+
+add_executable(${PROJECT} ${SOURCES})
```
Comment (oleksiibidnichenko) [inline src/processor_cxx11/CMakeLists.txt]: if you will define a project as mentioned above, you are able to use the ${CMAKE_PROJECT_NAME} variable offered by cmake itself.

File: src/processor_cxx11/main.cxx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main()
+{
+    std::cout << "I am processor with C++11..." << std::endl;
+    return 1;
```
Comment (oleksiibidnichenko) [inline src/processor_cxx11/main.cxx:6]: 0 indicates no errors (at least it is a default linux behavior)

File: src/processor_cxx17/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+set(PROJECT processor_cxx17)
```
Comment (oleksiibidnichenko) [inline src/processor_cxx17/CMakeLists.txt]: this is a redundant variable (see comment above)

File: src/processor_cxx17/main.cxx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main()
+{
+    std::cout << "I am processor with C++17..." << std::endl;
+    return 1;
```
Comment (oleksiibidnichenko) [inline src/processor_cxx17/main.cxx:6]: 0

File: src/traffic_generator/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+set(PROJECT traffic_generator)
```
Comment (oleksiibidnichenko) [inline src/traffic_generator/CMakeLists.txt]: this is a redundant variable (see comment above)

File: src/traffic_generator/main.cxx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main()
+{
+    std::cout << "I am traffic generator..." << std::endl;
+    return 1;
```
Comment (oleksiibidnichenko) [inline src/traffic_generator/main.cxx:6]: 0

File: src/processor_cxx11/main.cxx (hunk 3 above)
Comment (netch80) [reply src/processor_cxx11/main.cxx:6]: That's why I returned 1 - it does nothing and shall be fixed.

File: src/processor_cxx11/CMakeLists.txt (hunk 2 above)
Comment (netch80) [reply src/processor_cxx11/CMakeLists.txt]: This should be target - just typo. Will be fixed.

File: src/processor_cxx11/CMakeLists.txt (hunk 1 above)
Comment (netch80) [reply src/processor_cxx11/CMakeLists.txt]: Fixed.

File: src/processor_cxx11/CMakeLists.txt (hunk 2 above)
Comment (netch80) [reply src/processor_cxx11/CMakeLists.txt]: Fixed with PROJECT_NAME.

File: src/processor_cxx17/main.cxx (hunk 5 above)
Comment (netch80) [reply src/processor_cxx17/main.cxx:6]: Ditto.

**Review by oleksiibidnichenko — APPROVED** [review]

**Pull request merged**